hack/prepare-release.sh

```diff
@@ -0,0 +1,161 @@
+
+Example:
+
+  $this v0.15.0
+
+EOF
+}
+
+# Create temporary directory for GFD Helm chart
+create_gfd_helm_chart () {
+    rm -rf deployments/helm/gpu-feature-discovery
+    mkdir -p deployments/helm/gpu-feature-discovery
+    cp -r deployments/helm/nvidia-device-plugin/* deployments/helm/gpu-feature-discovery/
+    yq e -i '.devicePlugin.enabled = false | .gfd.enabled = true'  deployments/helm/gpu-feature-discovery/values.yaml
+    yq e -i '.name = "gpu-feature-discovery" | .description = "A Helm chart for gpu-feature-discovery on Kubernetes"' deployments/helm/gpu-feature-discovery/Chart.yaml
+}
+
+#
+# Parse command line
+#
+no_patching=
+no_assets=
+while getopts "abg:k:h" opt; do
+    case $opt in
+        a)  no_patching=y
+            ;;
+        b)  no_assets=y
+            ;;
+        h)  usage
+            exit 0
+            ;;
+        *)  usage
+            exit 1
+            ;;
+    esac
+done
+shift "$((OPTIND - 1))"
+
+# Check that no extra args were provided
+if [ $# -ne 1 ]; then
+    if [ $# -lt 1 ]; then
+        echo -e "ERROR: too few arguments\n"
+    else
+        echo -e "ERROR: unknown arguments: ${@:3}\n"
+    fi
+    usage
+    exit 1
+fi
+
+if [ -z "$no_patching" -a -z "$golang_version" ]; then
+    echo -e "ERROR: '-g GOLANG_VERSION' must be specified when modifying repo (i.e. when '-a' is not used)\n"
+    usage
+    exit 1
+fi
+
+release=$1
+shift 1
+
+container_image=nvcr.io/nvidia/k8s-device-plugin:$release
+
+#
+# Check/parse release number
+#
+if [ -z "$release" ]; then
+    echo -e "ERROR: missing RELEASE_VERSION\n"
+    usage
+    exit 1
+fi
+
+if [[ $release =~ ^(v[0-9]+\.[0-9]+)(\..+)?$ ]]; then
+    docs_version=${BASH_REMATCH[1]}
+    semver=${release:1}
+else
+    echo -e "ERROR: invalid RELEASE_VERSION '$release'"
+    exit 1
+fi
+
+#
+# Modify files in the repo to point to new release
+#
+if [ -z "$no_patching" ]; then
+    # Patch versions.mk
+    echo Patching versions.mk to refer to $release
+    sed -i.bak "s/^VERSION.*$/VERSION ?= $release/" versions.mk
+
+    # Patch deployments/static
+    echo Patching deployments to refer to $semver
+    find deployments/static -name "*.yaml" -exec sed -E -i '' \
+        -e s",^([[:space:]]+)app.kubernetes.io\/version:.+$,\1app.kubernetes.io\/version: $semver," \
+        -e s",^([[:space:]]+)- image:.+$,\1- image: $container_image," \
+        {} +
+    
+    # Patch deployments/helm/Chart.yaml
+    echo Patching deployments/helm/Chart.yaml to refer to $semver
+    sed -i.bak "s/^version: .*/version: \"$semver\"/" deployments/helm/nvidia-device-plugin/Chart.yaml
+    sed -i.bak "s/^appVersion: .*/appVersion: \"$semver\"/" deployments/helm/nvidia-device-plugin/Chart.yaml
+
+    # Patch README.md
+    # NOTE: this needs to be updated once we start releasing x.2.x versions
+    echo Patching README.md to refer to $release
+    sed -E -i.bak -e "s/v0\.[15-9]+\.[0-9]+/$release/g" README.md
+    sed -E -i.bak -e "s/0\.[15-9]+\.[0-9]+/$semver/g" README.md
```

Comment:
Note. This is usually a separate commit so that this can be cherry-picked onto the main branch, but maybe that's not important.

Also, why do we have to handle `release` and `semver` seprately? This match will also break once we release v1.

One more question: What does `[15-9]+` match?

Reply:
15-9 is to ignore some references to other dependencies 
examples 
line 
```md
* nvidia-docker >= 2.0 || nvidia-container-toolkit >= 1.7.0 (>= 1.11.0 to use integrated GPUs on Tegra-based systems)
```

or line 
```md 
As of now, the device plugin API for Kubernetes >= v1.10 is `v1beta1`.  If you
have a version of Kubernetes >= 1.10 you can deploy any device plugin version >
`v0.0.0`.
```

So this allow us to ignore those lines

Reply:
OK, but this breaks as soon as we want to replace `0.20.0`. Does it make sense to include the "from" verison in this replace rule?

Reply:
changed as per our offline discussion